blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineSearch.java

```diff
@@ -40,6 +44,7 @@
  */
 @Extension
 public class PipelineSearch extends OmniSearch<BluePipeline>{
+    private static final String STARTSWITH_PARAM="startsWith";
```

Comment:
@imeredith @NicuPascu  for example: if client wants to get all pipeline starting with 'blue' then it can send `/rest/search?q=type:pipeline;pipeline=^blue.*;...` then following code example can work:
```java
       String pipeline = q.param(getType()); //it should give string "^blue.*"
        if(pipeline != null) {
            Pattern pattern = Pattern.compile(pipeline);
            while (pipelineIterator.hasNext()) {
                BluePipeline p = pipelineIterator.next();
                if (!pattern.matches(pipeline).matches()) {
                    continue;
                }
                pipelines.add(p);
            }
            return Pageables.wrap(pipelines);
       }
```